src/nat.rs

```diff
@@ -32,8 +34,25 @@ pub fn get_public_ip_addr() -> Result<IpAddr, String> {
     }
 }
 
+pub fn udp_random_bind(start: u16, end: u16, mut tries: u32) -> Result<UdpSocket, String> {
+    tries -= 1;
+    let rand_port = thread_rng().gen_range(start, end);
+    let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), rand_port);
+
+    let bind_result = UdpSocket::bind(addr);
+    match bind_result {
+        Result::Ok(val) => Result::Ok(val),
+        Result::Err(err) =>
+            if tries > 0 {
+                udp_random_bind(start, end, tries)
+            } else {
+                panic!("Failed to bind to port {}, Error: {}", rand_port, err)
```

Comment:
No need to panic, just return the `Err(String)` to the caller and let them panic if they want